src/web/mod.rs

```diff
@@ -314,11 +319,11 @@ fn match_version(conn: &Connection, name: &str, version: Option<&str>) -> Option
     }
 
     // semver is acting weird for '*' (any) range if a crate only have pre-release versions
-    // return first version if requested version is '*'
-    if req_version == "*" && !versions_sem.is_empty() {
-        return Some(MatchVersion {
+    // return first non-yanked version if requested version is '*'
+    if req_version == "*" {
+        return versions_sem.first().map(|v| MatchVersion {
             corrected_name,
-            version: MatchSemver::Semver((versions_sem[0].0.to_string(), versions_sem[0].1)),
+            version: MatchSemver::Semver((v.0.to_string(), v.1)),
         });
     }
```

Comment:
Can you add a test that this returns a 404 if you request `/:crate` for a crate with only yanked versions?